### DISCUSSION TITLE
Composer require php pin dependencies is trying to use a none php 8.2.9 (not existing)

### DISCUSSION THREAD

**Discussion by rbel-io** [renovatebot/renovate#23858] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Pin dependencies is trying to update composer require to 8.2.9 which is a <strike>none</strike> existing PHP version.
The docker image does not exist in https://github.com/containerbase/php-prebuild/tags, version is build though in php source,
https://github.com/php/php-src/releases/tag/php-8.2.9

```
    "require": {
         "php": ">=8.1",
```

vs 

```
   "require": {
         "php": "8.2.9",
```

| Package | Type | Update | Change |
|---|---|---|---|
| [php](https://togithub.com/php/php-src) | require | pin | `>=8.1` -> `8.2.9` |

### Relevant debug logs

<details><summary>Logs</summary>

```
Command failed: install-tool php 8.2.9
Download failed: https://github.com/containerbase/php-prebuild/releases/download/8.2.9/php-8.2.9-focal-x86_64.tar.xz
[13:36:23.909] INFO (243): Downloading file ...
    url: "https://github.com/containerbase/php-prebuild/releases/download/8.2.9/php-8.2.9-focal-x86_64.tar.xz"
    output: "/tmp/renovate/cache/containerbase/feae542df538992a67bc38728369053a01fd10c59f6689e02ad1f0945f2126e6/php-8.2.9-focal-x86_64.tar.xz"
[13:36:24.208] FATAL (243): Response code 404 (Not Found)
    err: {
      "type": "HTTPError",
      "message": "Response code 404 (Not Found)",
      "stack":
          HTTPError: Response code 404 (Not Found)
              at _Request._onResponseBase (/snapshot/dist/containerbase-cli.js:39632:25)
              at _Request._onResponse (/snapshot/dist/containerbase-cli.js:39687:18)
              at ClientRequest.<anonymous> (/snapshot/dist/containerbase-cli.js:39702:17)
              at Object.onceWrapper (node:events:652:26)
              at ClientRequest.emit (node:events:549:35)
              at ClientRequest.emit (node:domain:482:12)
              at HTTPParser.parserOnIncomingClient (node:_http_client:642:27)
              at HTTPParser.parserOnHeadersComplete (node:_http_common:117:17)
              at TLSSocket.socketOnData (node:_http_client:505:22)
              at TLSSocket.emit (node:events:537:28)
      "code": "ERR_NON_2XX_3XX_RESPONSE",
```
</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
It seems that 8.2.9 has some problems or something unusual about it.

See https://github.com/php/php-src/issues/11868

**Comment by rbel-io** [comment]
Ok wow that's bad, any tips to (temporary) block PHP 8.2.9 from renovate config side?

**Reply by rarkins** [reply]
Set config.constraints.php = 8.2.8 in renovate config

**Reply by rbel-io** [reply]
This does not work unfortunately, although it's picked up in renovate as a config value.
It's still recreating the PR with the an upgrade to PHP 8.2.9.
What I'm missing here? 

p.s. I did close and rename the existing PR prior to the renovate run.


```
  "config": {
         "autodiscover": false,
         "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
         "onboarding": false,
         "platform": "github",
         "forkProcessing": "disabled",
         "prHourlyLimit": 0,
         "repositories": ["yyyy/renovate-xxx"],
         "labels": ["dependencies"],
         "constraints": {"php": "8.2.8"}
   }
```

**Comment by viceice** [comment, marked as answer]
https://hub.docker.com/_/php/tags?page=1&name=8.2.

8.2.9 is still a rc (pre-release) that's why it not yet on php-prebuild.

@rarkins we should use our php-prebuild as source here

https://github.com/renovatebot/renovate/blob/e61e3c80bcd00dbec0651d2b4cf097e6c0e85317/lib/modules/manager/composer/schema.ts#L274-L276

**Reply by rbel-io** [reply]
https://github.com/renovatebot/renovate/pull/23870/files

**Comment by rbel-io** [comment]
Fixed
